editor/plugins/shader_editor_plugin.cpp

```diff
@@ -37,12 +37,18 @@
 #include "editor/editor_node.h"
 #include "editor/editor_scale.h"
 #include "editor/editor_settings.h"
+#include "editor/project_settings_editor.h"
 #include "editor/property_editor.h"
 #include "servers/display_server.h"
 #include "servers/rendering/shader_types.h"
 
 /*** SHADER SCRIPT EDITOR ****/
 
+static bool saved_warnings_enabled = false;
+static bool saved_treat_warning_as_errors = false;
+static Map<ShaderWarning::Code, bool> saved_warnings;
+static uint32_t saved_warning_flags = 0U;
```

Comment:
Is there a specific reason to make those `static` here instead of private members of `ShaderTextEditor`?

Reply:
No, they will not work if I put them to class.

Reply:
Ah yeah you use them in both `ShaderEditor` and `ShaderTextEditor` I see.